driver-core/src/main/com/mongodb/internal/session/BaseClientSessionImpl.java

```diff
@@ -133,6 +134,21 @@ public void advanceClusterTime(final BsonDocument newClusterTime) {
         this.clusterTime = greaterOf(newClusterTime);
     }
 
+    @Override
+    public void setSnapshotTimestamp(final BsonTimestamp snapshotTimestamp) {
+        isTrue("open", !closed);
+        if (snapshotTimestamp == null) {
+            return;
```

Comment:
Don't overwrite existing value with null, since not all responses contain a snapshot timestamp.

Reply:
Simplification:
```
 if (snapshotTimestamp != null) {
    this.snapshotTimestamp = snapshotTimestamp;
 }
 ```

Reply:
I'm partial to early returns, but in this case not really useful.

Done